inc/core/MicroBitConfig.h

```diff
@@ -436,6 +452,34 @@ extern uint32_t __etext;
 #include "MicroBitHeapAllocator.h"
 #endif
 
+#if CONFIG_ENABLED(MICROBIT_IF_CHIP_PRESERVE)
+
+// Preserve our storage page used by MicroBitStorage, and the Bluetooth bonding
+// table.
+// Setting these values to 0 will disable preservation if this option is enabled.
+#ifndef MICROBIT_IF_PRES_1
+#define MICROBIT_IF_PRES_1                      0x0003BC00
+#endif
+
+#ifndef MICROBIT_IF_PRES_2
+#define MICROBIT_IF_PRES_2                      0x0003B800
+#endif
+
+// If we need to preserve pages, but we don't want the file system, then we
+// inform the constructor of JMXSerial to explicitly disable JMX, whilst preserving
+// the specific pages of flash memory above.
+
+// This ultimately means: pages can be preserved without the need (or overhead)
+// of file system request handlers.
+#if !CONFIG_ENABLED(MICROBIT_IF_CHIP_FS)
+#undef MICROBIT_IF_CHIP_FS
```

Comment:
Why is this pattern different to other options? why not just #ifndef MICROBIT_IF_CHIP_FS?